hearth/lib/hearth/http/header_list_builder.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Hearth
+  module Http
+    # @api private
+    module HeaderListBuilder
+      class << self
+        # builds a string from a list of possibly quoted values
+        # ensures that quoted values are escaped
+        # could name the below to be join_string_list?
+        def build_string_list(value)
+          unless value.respond_to?(:compact)
+            raise ArgumentError,
+                  "Expected #{value} to be an Array, got #{value.class} instead"
+          end
+
+          value.compact
+               .map { |s| escape_value(s) }
+               .join(', ')
+        end
+
+        private
+
+        # Escapes header field value
+        # @return [String]
```

Comment:
No need to document private methods with yard if you don't want to. It won't show up in documentation. Plus this method is self documenting - `escape_value(str)` essentially describes what it does and what it takes. In general, we strive to write self documenting code, instead of writing comments to describe code 💯